automate.py

```diff
@@ -164,6 +167,60 @@ def run_important_traces(self):
                 event.content['message'] = content
                 self._tracer.controller.buffers.app.put(event)
 
+    def schedule_id(self, id_):
+        """Keep track of scheduled ids"""
+        if not isinstance(id_, str):
+            raise AttributeError("Invalid id type, for schedule.")
+        self.ids.add(id_)
+        return id_
+
+    def schedule_traces(self, func, settings_=settings):
+        """Check for invalid arguments from schedule"""
+        if settings_.TRIGGER_SCHEDULE_TRACES:
+            id_ = self.schedule_id('automatic_traces')
+            trigger = settings_.SCHEDULE_TRIGGER
+            kwargs = settings_.SCHEDULE_ARGS
+            if (not isinstance(kwargs, dict) or
+                    not isinstance(trigger, str)):
+                raise AttributeError("Invalid attributes for "
+                                     "job to be scheduled.")
+            trigger_args = {}
+            trigger_args['kwargs'] = {'trigger': trigger}
+            trigger_args['kwargs'].update(kwargs)
+            self.scheduler.add_callable(id_, func, **trigger_args['kwargs'])
+            return self.scheduler.get_job(id_)
+        return None
+
+    def schedule_important_traces(self, func, settings_=settings):
+        """Check for invalid important arguments from schedule"""
+        if settings_.TRIGGER_IMPORTANT_CIRCUITS:
+            id_ = self.schedule_id('automatic_important_traces')
+            trigger = settings_.IMPORTANT_CIRCUITS_TRIGGER
+            kwargs = settings_.IMPORTANT_CIRCUITS_ARGS
+            if (not isinstance(kwargs, dict) or
+                    not isinstance(trigger, str)):
+                raise AttributeError("Invalid attributes for "
+                                     "job to be scheduled.")
+            trigger_args = {}
+            trigger_args['kwargs'] = {'trigger': trigger}
+            trigger_args['kwargs'].update(kwargs)
+            self.scheduler.add_callable(id_, func, **trigger_args['kwargs'])
+            return self.scheduler.get_job(id_)
+        return None
+
+    def unschedule_id(self, id_set):
+        """Remove ids to be unschedule"""
+        for id_ in id_set:
+            try:
+                self.ids.remove(id_)
```

Comment:
You can simplifying ids remove if you replace it with `self.ids.discard(id_)` no need to handle the exception since if it doesn't exist it won't raise `KeyError`.

Reply:
Oh, forgot about that alternative.